docs/signals/concepts/index.md

```diff
@@ -4,80 +4,224 @@ sidebar_position: 1
+* Events in real time (stream source only)
+* Events in warehouse (batch source only)
+* Interventions
+* Manually via Signals API
 
-## Interventions
+### Many-many relationship between entities and attributes
 
-Once an entity is defined in Signals you can start to retrieve interventions for it, and start publishing to them.
+A single entity will very likely have multiple attributes. For example, you could define a user entity with attributes for the number of page views, the last product they viewed, and their previous purchases.
 
-An <dfn>intervention</dfn> describes an action that can be performed for a user to achieve a more successful result.
+Imagined like this:
 
-User devices and your own systems can request interventions for a list of specific entities, which are then delivered in real-time as they are published.
-You can publish interventions manually using the API, or define them to trigger automatically when [attributes](#attributes) get updated and meet specific criteria.
-There are built-in operations that interventions can perform or that can be handled by Signals SDKs, but the contents of an intervention can contain custom data to use however you need to, and can include current attribute values for dynamic, personalized, actions using the latest real-time data.
+```python
+user.num_page_views_in_last_7_days
+user.last_product_viewed
+user.previous_purchases
+```
 
-For example, a user can subscribe to interventions for their own `domain_userid`, the current `app_id`, the current `page`, and the current `product`, and any interventions published targeting any of those entities get received.
-This enables both individual-level and broadcast-level messaging, so you can offer a specific user a personalized message, while also notifying all users on a specific product page that limited stock is selling fast.
+A single attribute definition could also be associated with multiple entities. For example, an attribute that counts page views might be relevant for user, page, and product entities. Attributes for different entities are distinct, even if they use the same name and definition.
 
-<!-- TODO: Read more about interventions -->
+Imagined like this:
+
+```python
+user.num_page_views_in_last_7_days
+page.num_page_views_in_last_7_days
+product.num_page_views_in_last_7_days
+```
+
+## Attributes can be grouped for ease of management
+
+### Views group attributes
+
+Configure attributes by grouping them into views. Each view is associated with a specific entity, source, version, and owner. It also has other optional metadata.
+
+Choose the source which fits your use case. For example, the attribute `last_product_viewed` is best calculated from events in-stream during a session, while `num_views_in_last_7_days` is best calculated from historical events.
+
+An example configuration for a view based on a user entity:
+
+```mermaid
+flowchart TD
+    User[User: `user_id`] --> Stream
+
+    subgraph Stream[Stream attributes]
```

Comment:
Also here it's a bit confusing to me that attributes are linked to the User entity and are defined as "Stream" – this is all part of the View definition, attributes don't define that, they are reusable across entities or stream/batch.